src/upnp.cpp

```diff
@@ -708,6 +708,22 @@ void upnp::post(upnp::rootdevice const& d, char const* soap
 #endif
 }
 
+std::string upnp::create_soap(char const* soap_action, std::string const& service_namespace, char const* part)
```

Comment:
for string :
span<char> create_xml_tag(std::initializer_list<std::pair<string_view, string_view>> const args)
{
	span<char> buf;
	for (auto const a : args)
	{
		int written = std::snprintf(buf.data(), buf.size(), "<%*s>%*s</%*s>\n"
			, int(a.first.size()), a.first.data()
			, int(a.second.size()), a.second.data()
			, int(a.first.size()), a.first.data());
		buf = buf.subspan(written);
	}
	return buf;
}
`
"<NewRemoteHost></NewRemoteHost>"
		"<NewExternalPort>%u</NewExternalPort>"
		"<NewProtocol>%s</NewProtocol>"
		"<NewInternalPort>%u</NewInternalPort>"
		"<NewInternalClient>%s</NewInternalClient>"
		"<NewEnabled>1</NewEnabled>"
		"<NewPortMappingDescription>%s at %s:%d</NewPortMappingDescription>"
		"<NewLeaseDuration>%u</NewLeaseDuration>"
`
%u  - unsigned int